turbo/jsonrpc/trace_adhoc.go

```diff
@@ -1286,8 +1289,6 @@ func (api *TraceAPIImpl) doCallMany(ctx context.Context, dbtx kv.Tx, msgs []type
 			finalizeTxStateWriter = noop
 		}
 
-		ibs.Reset()
```

Comment:
Removal of `Reset` here - probably will change `sd.CompareStates(initialIbs, ibs)`.
Example if you send 2 txs. 1st txn sends coin from A to B and from C to D, 2nd txn sends coin from B to A. Then state diff of 2nd txn may include `C to D` event too (even that 2nd txn didn't touch C and D accounts). 
Please add this case to unit-test.

Reply:
added test, check pls